src/main/java/org/folio/services/financedata/FinanceDataService.java

```diff
@@ -1,39 +1,206 @@
+import org.folio.rest.jaxrs.model.FyFinanceData;
 import org.folio.rest.jaxrs.model.FyFinanceDataCollection;
+import org.folio.rest.jaxrs.model.JobDetails;
+import org.folio.rest.jaxrs.model.Parameter;
+import org.folio.rest.jaxrs.model.Transaction;
+import org.folio.services.fiscalyear.FiscalYearService;
+import org.folio.services.fund.FundUpdateLogService;
 import org.folio.services.protection.AcqUnitsService;
+import org.folio.services.transactions.TransactionApiService;
 
 public class FinanceDataService {
+  private static final Logger log = LogManager.getLogger();
+
   private final RestClient restClient;
   private final AcqUnitsService acqUnitsService;
+  private final TransactionApiService transactionApiService;
+  private final FiscalYearService fiscalYearService;
+  private final FundUpdateLogService fundUpdateLogService;
 
-  public FinanceDataService(RestClient restClient, AcqUnitsService acqUnitsService) {
+  public FinanceDataService(RestClient restClient, AcqUnitsService acqUnitsService, TransactionApiService transactionApiService,
+                            FiscalYearService fiscalYearService, FundUpdateLogService fundUpdateLogService) {
     this.restClient = restClient;
     this.acqUnitsService = acqUnitsService;
+    this.transactionApiService = transactionApiService;
+    this.fiscalYearService = fiscalYearService;
+    this.fundUpdateLogService = fundUpdateLogService;
   }
 
   public Future<FyFinanceDataCollection> getFinanceDataWithAcqUnitsRestriction(String query, int offset, int limit,
                                                                                RequestContext requestContext) {
+    log.debug("Trying to get finance data with acq units restriction, query={}", query);
     return acqUnitsService.buildAcqUnitsCqlClauseForFinanceData(requestContext)
       .map(clause -> StringUtils.isEmpty(query) ? clause : combineCqlExpressions("and", clause, query))
       .compose(effectiveQuery -> getFinanceData(effectiveQuery, offset, limit, requestContext));
   }
 
-  private Future<FyFinanceDataCollection> getFinanceData(String query, int offset, int limit, RequestContext requestContext) {
+  private Future<FyFinanceDataCollection> getFinanceData(String query, int offset, int limit,
+                                                         RequestContext requestContext) {
     var requestEntry = new RequestEntry(resourcesPath(FINANCE_DATA_STORAGE))
       .withOffset(offset)
       .withLimit(limit)
       .withQuery(query);
     return restClient.get(requestEntry.buildEndpoint(), FyFinanceDataCollection.class, requestContext);
   }
+
+  public Future<Void> putFinanceData(FyFinanceDataCollection financeDataCollection, RequestContext requestContext) {
+    log.debug("Trying to update finance data collection with size: {}", financeDataCollection.getTotalRecords());
+    if (CollectionUtils.isEmpty(financeDataCollection.getFyFinanceData())) {
+      log.info("Finance data collection is empty, nothing to update");
```

Comment:
Method name can be included in the info log here

Reply:
added